### ISSUE TITLE
Single-line if breaks highlighting

### ISSUE THREAD

**Issue by rlivings39** [mathworks/MATLAB-Language-grammar#19]
```matlab
switch x
  case 1
    y = 1;
  case 2
    if cond, return; end
    y = 2;
  otherwise
    y = -1;
end
```

Reported in https://github.com/Gimly/vscode-matlab/issues/103 and https://github.com/Gimly/vscode-matlab/issues/101

**Comment by vr050714** [comment]
Actually, wrong coloring is in even simple case when `end` keyword appears on the same line with the last instruction:
```matlab
if condition
    do_something();  end  % <- there is problem
```

**Comment by rlivings39** [comment]
Reopening this issue as my changes in #22 introduced issues with matching `end` as a block end in too may places. For example:
```matlab
if cond
  y = x( end );
end
```
Treats the indexing `end` as the end of the `if`

**Comment by rlivings39** [comment]
Fixed by #22 and #28
